library/src/main/java/org/xmtp/android/library/Conversation.kt

```diff
@@ -28,154 +29,130 @@ sealed class Conversation {
     data class V2(val conversationV2: ConversationV2) : Conversation()
 
     data class Group(val group: org.xmtp.android.library.Group) : Conversation()
+    data class Dm(val dm: org.xmtp.android.library.Dm) : Conversation()
 
-    enum class Version { V1, V2, GROUP }
+    enum class Version { V1, V2, GROUP, DM }
 
-    // This indicates whether this a v1 or v2 conversation.
     val version: Version
         get() {
             return when (this) {
                 is V1 -> Version.V1
                 is V2 -> Version.V2
                 is Group -> Version.GROUP
+                is Dm -> Version.DM
+            }
+        }
+
+    val id: String
+        get() {
+            return when (this) {
+                is V1 -> throw XMTPException("Only supported for V3")
+                is V2 -> throw XMTPException("Only supported for V3")
+                is Group -> group.id
+                is Dm -> dm.id
```

Comment:
There are some places where querying by topic is still necessary. For example push notifications still expose only the topic. Since the topic just has the id inside it I have a method that parses the topic into the id to then search for it. In V2 mostly people were looking up conversations by address not topic.